src/overview/index.ts

```diff
@@ -45,32 +46,42 @@ export async function overviewCmdHandler(context: vscode.ExtensionContext, opera
 
   context.globalState.update(KEY_OVERVIEW_LAST_SHOW_TIME, Date.now().toString());
 
-  overviewView.iconPath = vscode.Uri.file(path.join(context.extensionPath, 'logo.lowres.png'));
-  let buffer = await readFile(path.join(context.extensionPath, './out/assets/overview/index.html'));
-  overviewView.webview.html = buffer.toString();
+  await initializeOverviewView(context, overviewView, onDidDisposeWebviewPanel);
+}
 
-  overviewView.onDidDispose(() => {
-    overviewView = undefined;
-  });
+function onDidDisposeWebviewPanel() {
+  overviewView = undefined;
+}
+
+async function initializeOverviewView(context: vscode.ExtensionContext, webviewPanel: vscode.WebviewPanel, onDisposeCallback: () => void) {
+  webviewPanel.iconPath = vscode.Uri.file(path.join(context.extensionPath, 'logo.lowres.png'));
+  webviewPanel.webview.html = await loadHtmlContent(context);
+
+  webviewPanel.onDidDispose(onDisposeCallback);
 
   const installedExtensions = vscode.extensions.all.map(ext => ext.id.toLowerCase());
-  overviewView.webview.postMessage({
+  webviewPanel.webview.postMessage({
     command: 'hideInstalledExtensions',
     installedExtensions: installedExtensions
   });
 
-  overviewView.webview.postMessage({
+  webviewPanel.webview.postMessage({
     command: 'setOverviewVisibility',
     visibility: context.globalState.get(KEY_SHOW_WHEN_USING_JAVA)
   });
 
-  overviewView.webview.onDidReceiveMessage((e) => {
+  webviewPanel.webview.onDidReceiveMessage((e) => {
     if (e.command === 'setOverviewVisibility') {
       toggleOverviewVisibilityOperation(context, e.visibility);
     }
   });
 }
 
+async function loadHtmlContent(context: vscode.ExtensionContext) {
+  let buffer = await readFile(path.join(context.extensionPath, './out/assets/overview/index.html'));
```

Comment:
I prefer to use `context.asAbsolutePath()` here. Using `path.join()` is fine also. Not a bit deal.

Reply:
This does produce cleaner code! thanks.